src/Interpreters/ClusterProxy/executeQuery.cpp

```diff
@@ -96,6 +97,14 @@ ContextMutablePtr updateSettingsForCluster(const Cluster & cluster, ContextPtr c
         new_settings.limit.changed = false;
     }
 
+    if (query_info && query_info->additional_filter_ast)
+    {
+        Tuple tuple;
+        tuple.push_back(main_table.getFullTableName());
+        tuple.push_back(queryToString(query_info->additional_filter_ast));
+        new_settings.additional_table_filters.value.push_back(std::move(tuple));
```

Comment:
Why it's not propagated automatically as every other query setting?

Reply:
It is propagated. But we need to change table name for distributed.